src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala

```diff
@@ -101,7 +104,7 @@ class CoreModuleImpl @Inject() (
     // infrastructure
     clock, random, marathonConf, actorSystem.scheduler,
     leadershipModule,
-    homeRegionProvider
+    () => marathonScheduler.getHomeRegion
```

Comment:
What motivated this change?

Reply:
@meln1k suggested it and I liked it... I would actually do it like this even from the beginning but then it was wired by guice and it was easier for me to introduce the interface.

You don't like it?

Reply:
I was just a little confused and too tired yesterday.